src/ansys_sphinx_theme/extension/linkcode.py

```diff
@@ -0,0 +1,253 @@
+"""Module containing ``link_code`` extension."""
+import inspect
+import os
+import os.path as op
+import sys
+
+from docutils import nodes
+from docutils.nodes import Node
+import sphinx
+from sphinx import addnodes
+from sphinx.application import Sphinx
+from sphinx.locale import _
+
+DOMAIN_KEYS = {
+    "py": ["module", "fullname"],
+    "c": ["names"],
+    "cpp": ["names"],
+    "js": ["object", "fullname"],
+}
+
+
+def sphinx_linkcode_resolve(
+    domain: str, info: dict, library: str, source_path: str, github_version: str, edit: bool = False
+) -> str or None:
+    """
+    Resolve the URL corresponding to a Python object for linking to the source code.
+
+    Parameters
+    ----------
+    domain : str
+        The domain of the object (e.g., 'py' for Python).
+
+    info : dict
+        A dictionary containing the information about the object.
+        It must have the keys 'module' and 'fullname'.
+
+    library : str
+        The repository/library name where the source code is hosted.
+        For example, 'ansys/ansys-sphinx-theme'.
+
+    source_path : str
+        The relative path of the source code file within the repository.
+        For example, 'src'.
+
+    github_version : str
+        The version of the library in github. It can be a specific version like
+        'main' or 'release/branch'.For versioned links, the version will be used in the URL.
+
+    edit : bool, default : False
+        If True, the link should point to the edit page for the source code.
+        Otherwise, it will point to the view page.
```

Comment:
No blank lines between documented parameters:

```suggestion
    Parameters
    ----------
    domain : str
        The domain of the object (e.g., 'py' for Python).
    info : dict
        A dictionary containing the information about the object.
        It must have the keys 'module' and 'fullname'.
    library : str
        The repository/library name where the source code is hosted.
        For example, 'ansys/ansys-sphinx-theme'.
    source_path : str
        The relative path of the source code file within the repository.
        For example, 'src'.
    github_version : str
        The version of the library in github. It can be a specific version like
        'main' or 'release/branch'.For versioned links, the version will be used in the URL.
    edit : bool, default : False
        If True, the link should point to the edit page for the source code.
        Otherwise, it will point to the view page.
```